sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldValueTypeInformation.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.utils.StaticSchemaInference.TypeInformation;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/** Represents type information for a schema field. */
+public class FieldValueTypeInformation implements Serializable {
+  private final String name;
+  private final Class type;
+  @Nullable private final Type elementType;
+  @Nullable private final Type mapKeyType;
+  @Nullable private final Type mapValueType;
+
+  public FieldValueTypeInformation(Field field) {
+    this(
+        field.getName(),
+        field.getType(),
+        getArrayComponentType(field),
+        getMapKeyType(field),
+        getMapValueType(field));
+  }
+
+  public FieldValueTypeInformation(TypeInformation typeInformation) {
+    this(
+        typeInformation.getName(),
+        typeInformation.getType().getRawType(),
+        getArrayComponentType(typeInformation),
+        getMapKeyType(typeInformation),
+        getMapValueType(typeInformation));
+  }
+
+  public FieldValueTypeInformation(
```

Comment:
private?

Reply:
no, used between schema and schema.utils.